src/object_store/src/object/s3.rs

```diff
@@ -958,12 +962,25 @@ impl tokio_retry::Condition<RetryError> for RetryCondition {
                         return true;
                     }
                 }
-                SdkError::ServiceError(e) => {
-                    if self.retry_unknown_service_error && e.err().code().is_none() {
+                SdkError::ServiceError(e) => match e.err().code() {
+                    None if self.retry_unknown_service_error => {
                         tracing::warn!(target: "unknown_service_error", "{e:?} occurs, retry S3 get_object request.");
                         return true;
                     }
-                }
+                    Some(code) if self.retry_slowdown_error => {
+                        if code == "SlowDown" {
+                            tracing::warn!(target: "slowdown_error", "{e:?} occurs, retry S3 get_object request.");
+                            return true;
+                        }
+                    }
+                    Some(code) if self.retry_too_many_requests_error => {
+                        if code == "TooManyRequests" {
+                            tracing::warn!(target: "too_many_requests_error", "{e:?} occurs, retry S3 get_object request.");
+                            return true;
+                        }
+                    }
```

Comment:
I am thinking maybe we should have a `Vec<String>` config to indicate the retryable error codes. Example:
```
[storage.object_store.s3]
retryable_service_error_codes = ["SlowDown", "TooManyRequests"]
```

In this case, user can modify the config accordingly based on the error codes of their state backend. What do you think?

Reply:
Makes sense. Updating it.